utils/writeable/src/helpers.rs

```diff
@@ -0,0 +1,106 @@
+// This file is part of ICU4X. For terms of use, please see the file
```

Comment:
This file contains helper code that I wanted to share between Writeable and TryWriteable.